ui/helpers/utils/settings-search.js

```diff
@@ -1,420 +1,54 @@
-      route: `${ABOUT_US_ROUTE}#links`,
-      icon: 'fa fa-info-circle',
-    },
-    {
-      tab: t('about'),
-      section: t('privacyMsg'),
-      description: t('privacyMsg'),
-      route: `${ABOUT_US_ROUTE}#privacy-policy`,
-      icon: 'fa fa-info-circle',
-    },
-    {
-      tab: t('about'),
-      section: t('terms'),
-      description: t('terms'),
-      route: `${ABOUT_US_ROUTE}#terms`,
-      icon: 'fa fa-info-circle',
-    },
-
-    {
-      tab: t('about'),
-      section: t('attributions'),
-      description: t('attributions'),
-      route: `${ABOUT_US_ROUTE}#attributions`,
-      icon: 'fa fa-info-circle',
-    },
-
-    {
-      tab: t('about'),
-      section: t('supportCenter'),
-      description: t('supportCenter'),
-      route: `${ABOUT_US_ROUTE}#supportcenter`,
-      icon: 'fa fa-info-circle',
-    },
+let settingsRoutes;
 
-    {
-      tab: t('about'),
-      section: t('visitWebSite'),
-      description: t('visitWebSite'),
-      route: `${ABOUT_US_ROUTE}#visitwebsite`,
-      icon: 'fa fa-info-circle',
-    },
-
-    {
-      tab: t('about'),
-      section: t('contactUs'),
-      description: t('contactUs'),
-      route: `${ABOUT_US_ROUTE}#contactus`,
-      icon: 'fa fa-info-circle',
-    },
-    {
-      tab: t('experimental'),
-      section: t('enableEIP1559V2'),
-      description: t('enableEIP1559V2Description'),
-      route: `${EXPERIMENTAL_ROUTE}#enable-advanced-gas`,
-      icon: 'fa fa-flask',
-    },
-  ];
-  if (process.env.TOKEN_DETECTION_V2) {
-    settingsRoutesList = [
-      ...settingsRoutesList,
-      {
-        /** TODO: Remove during TOKEN_DETECTION_V2 feature flag clean up */
-        tab: t('advanced'),
-        section: t('tokenDetection'),
-        description: t('tokenDetectionToggleDescription'),
-        route: `${ADVANCED_ROUTE}#token-description`,
-        icon: 'fas fa-sliders-h',
-      },
-    ];
-  } else {
-    settingsRoutesList = [
-      ...settingsRoutesList,
-      {
-        /** TODO: Remove during TOKEN_DETECTION_V2 feature flag clean up */
-        tab: t('experimental'),
-        section: t('useTokenDetection'),
-        description: t('useTokenDetectionDescription'),
-        route: `${EXPERIMENTAL_ROUTE}#token-description`,
-        icon: 'fa fa-flask',
-      },
-    ];
-  }
-  for (let i = 0; i < settingsRoutesList.length; i++) {
-    settingsRoutesList[i].id = i + 1;
+export function getSettingsRoutes() {
+  if (settingsRoutes) {
+    return settingsRoutes;
   }
-  // TODO: write to json file?
-  return showHideSettings(t, settingsRoutesList);
+  settingsRoutes = SETTINGS_ROUTES.filter((s) => {
+    if (s.featureFlag) {
+      return process.env[s.featureFlag];
+    }
+    return true;
+  });
+  return settingsRoutes;
 }
 
-function getFilteredSettingsRoutes(t, tabName) {
-  return getSettingsRoutes(t).filter((s) => s.tab === tabName);
+function getFilteredSettingsRoutes(tabName) {
+  return getSettingsRoutes().filter((s) => s.tab === tabName);
```

Comment:
Single-character variable names are bad and we should do something different, like in my suggestion above.